omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/EncounterProviderResource1_9.java

```diff
@@ -0,0 +1,134 @@
+            return description;
+        }
+        if(rep instanceof FullRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("provider",Representation.DEFAULT);
+            description.addProperty("encounter",Representation.DEFAULT);
+            description.addProperty("encounterRole",Representation.DEFAULT);
+            description.addProperty("voided");
+            description.addProperty("auditInfo",findMethod("getAuditInfo"));
+            description.addSelfLink();
+            return description;
+        }
+        return null;
+    }
+
+    @Override
+    public DelegatingResourceDescription getCreatableProperties() {
+        DelegatingResourceDescription description = new DelegatingResourceDescription();
+        description.addProperty("provider");
+        description.addProperty("encounter");
+        description.addProperty("encounterRole");
+        return description;
+    }
+
+    @Override
+    public DelegatingResourceDescription getUpdatableProperties() {
+        DelegatingResourceDescription description =  new DelegatingResourceDescription();
+        description.addProperty("encounterRole");
+        description.addProperty("voided");
+        description.addProperty("voidReason");
+        return description;
+    }
+
+    @Override
+    public Encounter getParent(EncounterProvider instance) {
+        return instance.getEncounter();
+    }
+
+    @Override
+    public void setParent(EncounterProvider instance, Encounter parent) {
+         instance.setEncounter(parent);
+    }
+
+    @Override
+    public PageableResult doGetAll(Encounter parent, RequestContext context) throws ResponseException {
+        List<EncounterProvider> encounterProviders = new ArrayList<EncounterProvider>(parent.getEncounterProviders());
+        return new NeedsPaging<EncounterProvider>(encounterProviders,context);
+    }
+
+    @Override
+    public EncounterProvider getByUniqueId(String uniqueId) {
+        return Context.getService(RestHelperService.class).getObjectByUuid(EncounterProvider.class,uniqueId);
+    }
+
+    @Override
+    protected void delete(EncounterProvider delegate, String reason, RequestContext context) throws ResponseException {
+         throw new ResourceDoesNotSupportOperationException();
+    }
+
+    @Override
+    public EncounterProvider newDelegate() {
+        return new EncounterProvider();
+    }
+
+    @Override
+    public EncounterProvider save(EncounterProvider delegate) {
+        delegate.getEncounter().getEncounterProviders().add(delegate);
+        return delegate;
+    }
+
+    @Override
+    public void purge(EncounterProvider delegate, RequestContext context) throws ResponseException {
+         delegate.getEncounter().removeProvider(delegate.getEncounterRole(),delegate.getProvider());
```

Comment:
Or may be I could move the functionality to delete() and throw exception in purge?